packages/runtime-core/src/apiLifecycle.ts

```diff
@@ -95,3 +96,31 @@ export const onErrorCaptured = (
 ) => {
   injectHook(LifecycleHooks.ERROR_CAPTURED, hook, target)
 }
+
+export function onServerPrefetch<
+  T extends () => Promise<any> = () => Promise<unknown>
+>(handler: T) {
+  const target = currentInstance
+  if (target) {
+    if (isInSSRComponentSetup) {
+      const type = target.type as ComponentOptions
+      let hook = type.serverPrefetch
+      if (hook) {
+        // Merge hook
+        type.serverPrefetch = () => handler().then(() => (hook as Function)())
```

Comment:
I think the `serverPrefect` hooks should be executed in parallel, because they are not related to each other, it is also important for performance, and conforms to the behavior of Vue2, and maybe we need to merge them into an array, and then deal with it here https://github.com/vuejs/vue-next/blob/master/packages/server-renderer/src/render.ts#L88, just wrap it with `Promise.all`. WDYT?